### DISCUSSION TITLE
custom proeprties / scalars

### DISCUSSION THREAD

**Discussion by nrbnlulu** [geldata/gel#5254] [Q&A] [answered]
What would be EdgeDb alternative implementation for custom fields?
i.e [django-phonenumber-field](https://github.com/stefanfoulis/django-phonenumber-field)

**Comment by raddevon** [comment, marked as answer]
Your best bet is probably to use [a custom scalar with a regex constraint](https://www.edgedb.com/docs/datamodel/constraints#constraints-on-custom-scalars).

**Reply by nrbnlulu** [reply]
I see... so generally you would need to use something like Pydantic  more complex use cases?

**Reply by raddevon** [reply]
Sure, you could do something more complex in your application. You could also use access policies to create [super constraints](https://www.edgedb.com/docs/datamodel/access_policies#super-constraints) or you might be able to use our upcoming [triggers](https://github.com/edgedb/rfcs/blob/master/text/1020-triggers.rst) feature if you need something even more robust and want to keep it in the database instead of in your application.
